edb/edgeql/compiler/viewgen.py

```diff
@@ -490,6 +490,15 @@ def _normalize_view_ptr_expr(
             name=ptrcls.get_shortname(ctx.env.schema).name,
         )
 
+        # Schema computables that point to opaque unions will just have
+        # BaseObject as their target, but in order to properly compile
+        # it, we need to know the actual type here, so we recompute it.
+        is_opaque_schema_computable = (
+            ptrcls.is_pure_computable(ctx.env.schema)
+            and (t := ptrcls.get_target(ctx.env.schema))
```

Comment:
This is one of those things my computable/alias rework is fixing.  Let's put an XXX here to remove the hack once that lands.